plugins/become/pfexec.py

```diff
@@ -101,4 +101,4 @@ def build_become_command(self, cmd, shell):
 
         flags = self.get_option('become_flags')
         noexe = not self.get_option('wrap_exe')
-        return '%s %s "%s"' % (exe, flags, self._build_success_command(cmd, shell, noexe=noexe))
+        return '%s %s %s' % (exe, flags, self._build_success_command(cmd, shell, noexe=noexe))
```

Comment:
```suggestion
        return '%s %s %s' % (exe, flags, shlex.quote(self._build_success_command(cmd, shell, noexe=noexe)))
```

I don't have pfexec so can't test.

Reply:
@manschwetusCS did you take a look at this suggestion?

Reply:
I guess you mean `shlex_quote(...)`, tried it right now, it does not work.
```
...
return '%s %s %s' % (exe, flags, shlex_quote(self._build_success_command(cmd, shell, noexe=noexe)))
...
```

```
HOST  | FAILED! => {
    "changed": false,
    "module_stderr": "Shared connection to HOST closed.\r\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

Reply:
@felixfontein @Akasurde would you vote this as resolved?

Reply:
@manschwetusCS What happens when the cmd has spaces?

Reply:
As for illumos wrap_exe is needed, it works.  
Also self._build_success_command, shlex-quotes the entire command anyway, so maybe double shlex-quoting gets too much.  
An option would be to keep it when noexe is given.

Reply:
I looked a lot at this and other become plugins, including the ones in core, and I think this is correct.